### DISCUSSION TITLE
pushing routes with `queryParams`

### DISCUSSION THREAD

**Discussion by raulmabe** [csells/go_router#88] [Q&A] [answered]
I am trying to push a route (`/search`) with `queryParams` from a top-level route (`/`) but throws `AssertionError` because of the following constraint `assert(identical(match.queryParams, matches.first.queryParams));` in [line 661](https://github.com/csells/go_router/blob/master/lib/src/go_router_impl.dart#L661).

Is it incompatible to use push and `queryParams` at the same time? Same happens with the `extra` property

**Comment by csells** [comment, marked as answer]
yep. this is "supported" in the sense that it should work. the lack of support is a bug that I'm working on.
